docs/api/design.md

```diff
@@ -0,0 +1,528 @@
+# Waddle-API Design Document
+
+## Overview
+This document outlines the API specification for a podcast editing web application backend that integrates with the Waddle preprocessing library. The system will handle audio file processing, transcription, editing, and final export functionality including metadata and chapter generation.
+
+## API Endpoints
+
+### 1. Episode Management
+
+#### 1.1 Create Episode
+- **Endpoint**: `POST /api/episodes`
+- **Description**: Creates a new episode and initiates preprocessing
+- **Request Body**: `EpisodeCreate` model
+  ```typescript
+  {
+    title: string;
+    editor_state?: string;  // Optional initial editor state
+    preprocessed?: boolean; // Default: false
+    postprocessed?: boolean; // Default: false
+  }
+  ```
+- **Request Files**: Audio files (multipart/form-data)
+- **Response**: `Episode` model
+
+#### 1.2 Get Episode
+- **Endpoint**: `GET /api/episodes/{uuid}`
+- **Description**: Retrieves episode metadata and status
+- **Response**: `Episode` model (same as Create Episode response)
+
+#### 1.3 Update Episode
+- **Endpoint**: `PUT /api/episodes/{uuid}`
+- **Description**: Updates episode metadata and editor state
+- **Request Body**: `EpisodeUpdate` model
+  ```typescript
+  {
+    title?: string;
+    editor_state?: string;
+    preprocessed?: boolean;
+    postprocessed?: boolean;
+    metadata_generated?: boolean;
+  }
+  ```
+- **Response**: `Episode` model
+
+### 2. Audio Processing
```

Comment:
I considered using `GET /api/episodes/{uuid}/preprocessed-audios` and `GET /api/episodes/{uuid}/preprocessed-audio/{file_id}`, but since there is no need to access raw audio, I decided to use `audio`. However, adding `preprocessed` might still be an option for clarity. Let me know your thoughts!